### ISSUE TITLE
Add Filters by Donor, Grant and Location on the PD Screen (5.5)

### ISSUE THREAD

**Issue by mpawlowski85** [unicef/etools-issues#996]
The List of Programme Documents/SSFA should have additional filters added. These are: 
-	Grant: while grant information is not displayed in eTools, the information is available at the FR line item level and it is loaded to eTools databases. One Grant can fund multiple FRs. When a particular Grant is selected, all of the PDs that have FRs associated with that particular grant should be selected. The Grants should be listed by “Grant Number” in the filter menu. A user should be able to select multiple Grants from the filter.
-	Donor: while Donor information is not displayed in eTools, the information is available in the “Grant” API and it is loaded to eTools. Each Grant is associated with a Donor and each FR is associated with a Grant, thus there is an indirect connection between an FR and Donor. When selecting a Donor from the filter list, all of the PDs that have the FRs that are funded by that Donor should be displayed. A user should be able to select multiple Donors from the filter. 

On the PD page, the Grant and Donor should be displayed after an FR is added to the PD/SSFA. 
@evrimm to provide wireframes for the changes.

**Comment by robertavram** [comment]
we need to split this in two tasks... 1 BE, 1 FE.. FE depends on BE...

**Comment by adi130987** [comment]
FE 13pts

**Comment by robertavram** [comment]
filters on BE?

**Comment by robertavram** [comment]
Donors and grants.. BE vision synchronizers need to be adapted and tested to include these relationships + address performance concerns. Serializers + moving filtering/sorting on BE. BE estimation excluding locations is 20. @ntrncic 

...Not sure how the location filter will work.. it seems like an export with a "pivoted view" on location is desired... not sure if a filter in the frontend is desired also... please let us know.

**Comment by evrimm** [comment]
@mpawlowski85 When you refer to PD page where you requested to show Donor and Grant info after FR is added, do you mean showing them within the Fund Reservations card next to FR number? If so, we have to show each line item for that specific fund reservation header number in order to show all grants. 
While looking into this I realized that although we had revisions to FR card a few months back (mockups here: https://invis.io/7CCGJCJK6#/242285298_03_-_Add_PD-SSFA_ToR_-_Review_-_Sign  ) Did we revert from this view?

**Comment by mpawlowski85** [comment]
@evrimm 
eTools will only know the Donor and Grant information once the FR number is added - hence the display only after the FR number is sucesfully added. 
The information about the donor and the grant should be displayed in the PD Overview. Would it be possible to mock this up? 
We might also need this info in the partner overview, but will confirm this  later.

**Comment by evrimm** [comment]
@mpawlowski85 I was thinking displaying it in the Fund Reservation card under Review and Sign as there is quiet a bit of space as there is nothing except the FR number. It would be useful for the user to see some more data about FR number as a confirmation that they have entered the right FR number. 
In the PD overview, we call the card Implementation Status and show how much of the FR has been utilized. If we show Donor, Grant information here, we need to rethink the entire card because there is one to many relationship between FR number and grants.
In short, my recommendation is to put this info in FR card under Review and Sign.
